crates/wasi-common/src/sys/unix/linux/filetime.rs

```diff
@@ -0,0 +1,53 @@
+use super::super::filetime::FileTime;
+use std::fs::File;
+use std::io;
+use std::sync::atomic::{AtomicBool, Ordering::SeqCst};
+
+pub(crate) const UTIME_NOW: i64 = libc::UTIME_NOW;
+pub(crate) const UTIME_OMIT: i64 = libc::UTIME_OMIT;
+
+pub(crate) fn utimensat(
+    dirfd: &File,
+    path: &str,
+    atime: FileTime,
+    mtime: FileTime,
+    symlink: bool,
+) -> io::Result<()> {
+    use super::super::filetime::{to_timespec, utimesat};
+    use std::ffi::CString;
+    use std::os::unix::prelude::*;
+
+    let flags = if symlink {
+        libc::AT_SYMLINK_NOFOLLOW
```

Comment:
Is the sense here reversed? It looks like `symlink` here means *do* follow, but the flag says *no* follow.

Reply:
Ooops, that's my bad. This bit of code is correct, but the one in `path_filestat_set_times` should set the `bool` flag the opposite way, i.e.,

```
let symlink_nofollow = wasi::__WASI_LOOKUP_SYMLINK_FOLLOW != dirflags;
```

I've fixed it now and renamed all `symlink` occurrences with `symlink_nofollow` for easier tracking. Anyhow, that was a good catch, thanks!